doc/mutations/creating-mutation-plugins.md

```diff
@@ -1,7 +1,192 @@
 
-WIP : For now for relevant information on custom mutations check the following links : 
+## Why the automatic mutations where removed. 
 
--  https://www.amazeelabs.com/en/blog/extending-graphql-part-3-mutations
--  https://github.com/justinlevi/graphql_custom_mutation
--  http://joaogarin.com/posts/drupal-graphql-with-angular-and-apollo-part3
+In [this article](https://www.amazeelabs.com/en/blog/extending-graphql-part-3-mutations) it's well explained why automatic mutations where removed. But this, as stated in the article, does not mean that creating mutations is complicated. In fact, its a simple task and one that might even provide with the extra flexibility you know and love from Drupal.
+
+## Mutations to create Drupal Entities
+One of the most common things you will want to do when with mutations is to create, update and delete entities. These CRUD operations on entities where made as simple as possible to implement with the module by extending some generic classes provided by the graphql_core module.
+
+So lets have a look at how you can create a mutation from scratch to generate a new entity of type node, and in this case a new article. You can refer to the [Examples](https://github.com/drupal-graphql/graphql-examples) repository to look at some other examples as well for how to create other kinds of mutations.
+
+### CreateArticle Plugin
+
+The first step to create a mutation is to make the plugin, the graphql module provides a base class for creating new entities called **CreateEntityBase**. You should implement a plugin that extends this class when you want to create an entity.
+
+lets look at what the code for this plugin looks like : 
+
+```
+<?php
+namespace Drupal\graphql_examples\Plugin\GraphQL\Mutations;
+use Drupal\graphql\Annotation\GraphQLMutation;
+use Drupal\graphql\GraphQL\Execution\ResolveContext;
+use Drupal\graphql_core\Plugin\GraphQL\Mutations\Entity\CreateEntityBase;
+use GraphQL\Type\Definition\ResolveInfo;
+/**
+ * Simple mutation for creating a new article node.
+ *
+ * @GraphQLMutation(
+ *   id = "create_article",
+ *   entity_type = "node",
+ *   entity_bundle = "article",
+ *   secure = true,
+ *   name = "createArticle",
+ *   type = "EntityCrudOutput!",
+ *   arguments = {
+ *     "input" = "ArticleInput"
+ *   }
+ * )
+ */
+class CreateArticle extends CreateEntityBase {
+  /**
+   * {@inheritdoc}
+   */
+  protected function extractEntityInput(
+    $value,
+    array $args,
+    ResolveContext $context,
+    ResolveInfo $info
+  ) {
+    return [
+      'title' => $args['input']['title'],
+      'body' => $args['input']['body'],
+    ];
+  }
+}
+```
+
+
+We can see a couple things here in this code that are particular interesting : 
```

Comment:
particularly

Reply:
Redundant whitespace before the colon

Reply:
Also, the 'here' seems redundant. "We can see a couple things in this code that are particularly interesting:"